app/models/manageiq/providers/vmware/network_manager/refresh_parser.rb

```diff
@@ -0,0 +1,207 @@
+module ManageIQ::Providers
+  class Vmware::NetworkManager::RefreshParser
+    include ManageIQ::Providers::Vmware::RefreshHelperMethods
+
+    def initialize(ems, options = nil)
+      @ems        = ems
+      @connection = ems.connect
+      @options    = options || {}
+      @data       = {}
+      @data_index = {}
+      @inv        = Hash.new { |h, k| h[k] = [] }
+      @org        = @connection.organizations.first
+    end
+
+    def ems_inv_to_hashes
+      log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{@ems.name}] id: [#{@ems.id}]"
+
+      $log.info("#{log_header}...")
+
+      get_networks
+      get_network_ports
+
+      $log.info("#{log_header}...Complete")
+
+      @data
+    end
+
+    private
+
+    def get_networks
+      # fetch org VDC networks
+      @inv[:networks] = @org.networks
+
+      # fetch vApp networks
+      @inv[:networks] += fake_vapp_networks
+
+      process_collection(@inv[:networks], :cloud_networks) { |n| parse_network(n) }
+      process_collection(@inv[:networks], :cloud_subnets) { |n| parse_network_subnet(n) }
+    end
+
+    def get_network_ports
+      # TODO: Update FOG so that it will provide list of network ports. Currently, it is only possible to obtain
+      # very little data, barely enough to tell to which network the VM is connected to.
+      @inv[:vm_networks] = []
+      @org.vdcs.each do |vdc|
+        vdc.vapps.each do |vapp|
+          vapp.vms.each do |vm|
+            @inv[:vm_networks] += vm.network_adapters.each_with_index.map do |net_if, i|
+              net_if[:vapp] = vapp
+              net_if[:vm] = vm
+              net_if[:vm_network] = vm.network
+              net_if[:idx] = i
+              net_if
+            end
+          end
+        end
+      end
+
+      $log.info("... captured vm_networks with foggy: #{@inv[:vm_networks]}")
```

Comment:
Was this for debugging?  Not sure we want to dump all the vm_networks to the log